### PR TITLE
docs(*): add initial cheatsheet tags

### PR DESCRIPTION
**Pull request by petebacondarwin**
This PR is to show @naomiblack what the tags need to look like for the angular.io cheatsheet.

### CODE REVIEW

**Comment by IgorMinar** [comment]
have we considered building up the cheatsheet from the first paragraph of each regular doc marked with `@cheatsheetAble` annotation?

alternatively, split the large doc chunks into small doc annotations that are part of regular docs?

**Comment by petebacondarwin** [comment]
@IgorMinar I am open to other ideas.

Currently the cheatsheet sheet consists of a set of sections, each with a name, description and a list of items. Each item consists of a piece of syntax a description and a collection of pieces of the syntax that need to be highlighted in bold.

The items don't map 1-1 with docs so I thought it was easier to map sections to docs instead.

We could easily put the cheatsheet in its own set of specialised docs but I think there is value in keeping cheatsheet items near to the place their syntax is defined.

File: modules/angular2/src/core/application.ts [hunk 1]
```diff
@@ -18,6 +18,61 @@ export {
...
+ *           #movieplayer
+ *           (click)
+ * @cheatsheetItem
+ *          <p *my-unless="myExpression">...</p>
+ *          The * symbol means that the current element will be turned into an embedded template. Equivalent to:\n<template [my-unless]="myExpression"><p>...</p></template>
+ *          *my-unless
+ * @cheatsheetItem
+ *          <p>Card No.: {{cardNumber myCreditCardNumberFormatter}}</p>
+ *          Transforms the current value of expression cardNumber via pipe called creditCardNumberFormatter.
+ *          {{cardNumber myCreditCardNumberFormatter}}
```
Comment (todoubaba) [inline modules/angular2/src/core/application.ts]: {{cardNumber | myCreditCardNumberFormatter}}

Comment (petebacondarwin) [inline modules/angular2/src/core/application.ts]: Thanks

**Comment by petebacondarwin** [comment]
OK, so I have refactored this. Now the cheatsheet sections are in their own files under the ` modules/angular2/docs/cheatsheet` folder. This seems about the right place to keep them - Not too far from or too close to the source - and fairly clean layout. See the updated PR at https://github.com/angular/angular.io/pull/348 for more info

**Comment by petebacondarwin** [comment]
The format of the cheatsheet section files is as follows:

* `@cheatsheetSection some title text` indicates that this is a cheatsheet section and provides its title
* `@description some addition markdown rendered content` is the additional description of the section
* `@cheatsheetIndex 3` indicates where in the cheatsheet this section should appear
* `@cheatsheetItem` contains information about a particular item. The item consists of a number of blocks of text:
  * The "syntax" block comes first and must be wrapped in backticks
  * Next there are zero or more "bold matcher" blocks, which are separated by a pipe `|` and each must also be wrapped in backticks
  * Finally the rest of the text is the description, which will be rendered as markdown

**Comment by naomiblack** [comment]
I like this. It's pretty clean, and I like getting this out of the main code files. Much tidier. @IgorMinar I disagree on just using the first line of comments in the API -- requires too much vigilance to make sure everyone adheres to style, and we lose the ability to make it specific to the example itself. LGTM

**Comment by naomiblack** [comment]
Chatted with Igor -- go ahead with this proposal!

**Comment by petebacondarwin** [comment]
I am guessing that this will close when it lands in master?

**Comment by petebacondarwin** [comment]
Oh no. I see it is there already: E.g. https://github.com/angular/angular/commit/9b3b721f08e45d2b8224630bb82662652b1ef0b4

**Pull request closed without merging**